### ISSUE TITLE
Request Off-The-Record feature setting stuck on 'Always go' 

### ISSUE THREAD

**Issue by Uni-verse** [brave/brave-browser#31857]
## Description 
Attempting to set Request Off The Record to one of the other options does not take effect

## Steps to reproduce 

   1. Install `1.58.43, Chromium 115.0.5790.136`
   2. Open settings -> Brave Shields & Privacy > Request Off-The-Record (other privacy settings section)
   3. Change from 'Always on' to 'Ask before going'
   4. Tap `Done`
   5. Observe setting preference

## Actual result 
Setting did not change
![Screenshot 2023-07-25 at 3 09 15 PM](https://github.com/brave/brave-browser/assets/17885425/a6113595-d41a-4585-b2ed-63417386dfa6)

## Expected result
Should change

## Issue reproduces how often 
Always

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? No
- Can you reproduce this issue with the current Play Store Beta version? No
- Can you reproduce this issue with the current Play Store Nightly version? Yes

## Device details
   - Install type (ARM, x86): ARM x64
   - Device type (Phone, Tablet, Phablet): Samsung GS 21
   - Android version: 13
   
## Brave version
1.58.43, Chromium 115.0.5790.136
1.58.44, Chromium 116.x - reproducible

**Comment by anthonypkeane** [comment]
Can you pls link to the original issue here Deep for reference. Thanks

**Comment by deeppandya** [comment]
@anthonypkeane work was done here : https://github.com/brave/brave-core/pull/18931. i don't see any issue attached to the PR.

**Comment by kjozwiak** [comment]
The above requires `1.58.98` or higher for `1.58.x` verification 👍

**Comment by anthonypkeane** [comment]
@deeppandya I don't see the Android UI for this on https://github.com/brave/brave-core/pull/18931, do you know where it is?

**Comment by deeppandya** [comment]
> @deeppandya I don't see the Android UI for this on https://github.com/brave/brave-core/pull/18931, do you know where it is?

@anthonypkeane we added new settings option in shields. ![Screenshot_20230823-131628.png](https://github.com/brave/brave-browser/assets/5146988/a1fd7ab4-6e08-4a80-acc9-2bfd63776032)

![Screenshot_20230823-131624.png](https://github.com/brave/brave-browser/assets/5146988/87f81ee8-cb14-4329-829c-87f58fa9b6c7)

**Comment by anthonypkeane** [comment]
If I go off the record, what is the UI in the app, on a webpage?

**Comment by anthonypkeane** [comment]
Needs UI before we can move this to Release

**Comment by anthonypkeane** [comment]
The UI for this isn't correct. We need a follow up issue asap to resolve the inconsistencies with the UI vs the rest of the app

**Comment by kjozwiak** [comment]
@anthonypkeane going to close the above as we usually don't re-open issues. Can you create a new issue re: updating the design? We also have https://github.com/brave/brave-browser/issues/32655 to address the issue not working as expected when visiting a website with OTR enabled. The above was also removed as per https://github.com/brave/brave-browser/issues/32501.
